content/storage/docs/spark/instance-store.md

```diff
@@ -0,0 +1,146 @@
+      - "mount -a"
+      - "chown 999:1000 /spark_data_*"
+```
+
+
+In order to successfully use these volumes with Spark, it’s required to specify additional configurations to use these volumes during shuffles operations. To use a volume as local storage, the volume’s name should starts with `spark-local-dir-`.
+
+Below an example configuration provided during the EMR on EKS job submission, that shows how to configure Spark to use 2 volumes as local storage for the job.
+
+**Spark Configurations**
+
+```
+{
+  "name": ....,
+  "virtualClusterId": ....,
+  "executionRoleArn": ....,
+  "releaseLabel": ....,
+  "jobDriver": ....,
+  "configurationOverrides": {
+    "applicationConfiguration": [
+      {
+        "classification": "spark-defaults",
+        "properties": {
+          "spark.kubernetes.executor.volumes.hostPath.spark-local-dir-1.mount.path": "/spark_data_1",
+          "spark.kubernetes.executor.volumes.hostPath.spark-local-dir-1.mount.readOnly": "false",
+          "spark.kubernetes.executor.volumes.hostPath.spark-local-dir-1.options.path": "/spark_data_1",
+          "spark.kubernetes.executor.volumes.hostPath.spark-local-dir-2.mount.path": "/spark_data_2",
+          "spark.kubernetes.executor.volumes.hostPath.spark-local-dir-2.mount.readOnly": "false",
+          "spark.kubernetes.executor.volumes.hostPath.spark-local-dir-2.options.path": "/spark_data_2"
+        }
+      }
+    ]
+  }
+}
+```
+
+
+Please note that for this approach it is required to specify the following configurations for each volume that you want to use. (IDX is a label to identify the volume mounted)
+
+```
+# Mount path on the host node
+spark.kubernetes.executor.volumes.hostPath.spark-local-dir-IDX.options.path
+
+# Mount path on the k8s pod
+spark.kubernetes.executor.volumes.hostPath.spark-local-dir-IDX.mount.path
+
+# (boolean) Should be defined as false to allow Spark to write in the path
+spark.kubernetes.executor.volumes.hostPath.spark-local-dir-IDX.mount.readOnly
+```
+
+
+**Benefits**
+
+* Possible to re-partition volumes across different Spark jobs (e.g. a node group containing instances with 2 instance volumes each, can dedicate one volume per Spark job)
+
+**Cons**
+
+* Additional configurations required for Spark jobs to use individual volumes. This approach can be error prone if you don’t control the instance types where your jobs are launched (e.g. multiple node groups with different instance types). In this case can be useful to specify the following spark configuration to launch the executors on a specific instance type: **[spark.kubernetes.node.selector.node.kubernetes.io/instance-type](http://spark.kubernetes.node.selector.node.kubernetes.io/instance-type)**
+* Data created on the volumes is automatically deleted at job completion. However, in case of failures, data might be retained in the disks.
```

Comment:
Data created on the volumes is automatically deleted once the job is completed and instance is terminated. However, you need to extra measures to delete the data on instance store volumes if EC2 instance is re-used or is not terminated.